redfish-core/lib/pcie.hpp

```diff
@@ -124,73 +124,158 @@ inline void requestRoutesSystemPCIeDevice(App& app)
+                                    *property;
+                            }
                         }
-                    }
 
-                    asyncResp->res.jsonValue["PCIeFunctions"] = {
-                        {"@odata.id",
-                         "/redfish/v1/Systems/system/PCIeDevices/" + device +
-                             "/PCIeFunctions"}};
-                };
+                        if (std::string* property = std::get_if<std::string>(
+                                &pcieDevProperties["Model"]);
+                            property)
+                        {
+                            if (!property->empty())
+                            {
+                                asyncResp->res.jsonValue["Model"] = *property;
+                            }
+                        }
+
+                        if (std::string* property = std::get_if<std::string>(
+                                &pcieDevProperties["SparePartNumber"]);
+                            property)
+                        {
+                            if (!property->empty())
+                            {
+                                asyncResp->res.jsonValue["SparePartNumber"] =
+                                    *property;
+                            }
+                        }
+
+                        if (std::string* property = std::get_if<std::string>(
+                                &pcieDevProperties["PrettyName"]);
+                            property)
+                        {
+                            if (!property->empty())
+                            {
+                                asyncResp->res.jsonValue["Name"] = *property;
+                            }
+                        }
+
+                        if (std::string* property = std::get_if<std::string>(
+                                &pcieDevProperties["LocationCode"]);
+                            property)
+                        {
+                            if (!property->empty())
+                            {
+                                asyncResp->res
+                                    .jsonValue["Slot"]["Location"]
+                                              ["PartLocation"]["ServiceLabel"] =
+                                    *property;
+                            }
+                        }
+
+                        if (bool* property = std::get_if<bool>(
+                                &pcieDevProperties["Present"]);
+                            property)
+                        {
+                            if (*property == true)
+                            {
+                                asyncResp->res.jsonValue["Status"]["State"] =
+                                    "Enabled";
+                            }
+                            else
+                            {
+                                asyncResp->res.jsonValue["Status"]["State"] =
```

Comment:
We typically have been doing more like a default state (e..g . ["Status"]["State"] = Enabled) outside of the call and then in here only have 
                             if (*property == false)
                             {
                                asyncResp->res.jsonValue["Status"]["State"] = "Absent";
                             
                             }